internal/osutil/mkdirall_windows.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2014 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package osutil
+
+import (
+	"os"
+	"syscall"
+)
+
+// See https://github.com/golang/go/issues/10900
+
+// MkdirAll creates a directory named path,
+// along with any necessary parents, and returns nil,
+// or else returns an error.
+// The permission bits perm are used for all
+// directories that MkdirAll creates.
+// If path is already a directory, MkdirAll does nothing
+// and returns nil.
+func MkdirAll(path string, perm os.FileMode) error {
+	// Fast path: if we can tell whether path is a directory or file, stop with success or error.
+	dir, err := os.Stat(path)
+	if err == nil {
+		if dir.IsDir() {
+			return nil
+		}
+		return &PathError{"mkdir", path, syscall.ENOTDIR}
+	}
+
+	// Slow path: make sure parent exists and then call Mkdir for path.
+	i := len(path)
+	for i > 0 && os.IsPathSeparator(path[i-1]) { // Skip trailing path separator.
+		i--
+	}
+
+	j := i
+	for j > 0 && !os.IsPathSeparator(path[j-1]) { // Scan backward over element.
+		j--
+	}
+
+	if j > 1 {
+		// Create parent
+		parent := path[0 : j-1]
+		if parent != os.VolumeName(parent) {
+			err = MkdirAll(parent, perm)
```

Comment:
I'm not super fond of the recursiveness here... How about some like ... (pseudocode)

```go
cleaned := filepath.Clean(path)
if strings.HasPrefix(cleaned, `\\?\`) { cleaned = cleaned[4:] }
split := strings.Split(cleaned, os.PathSeparator)
for i := range split {
  path := `\\?\` + strings.Join(split[:i], os.PathSeparator)
  if os.Stat(path) is happy { continue }
  err := os.Mkdir(path)
 ...
}
```
?